src/shared/cloudformation/templateRegistry.ts

```diff
@@ -28,50 +27,10 @@ export function setTemplateRegistry(registry: CloudFormationTemplateRegistry | u
 export interface CloudFormationTemplateRegistry {
     registeredTemplates: Map<string, CloudFormation.Template>
     getRegisteredTemplate(templatePath: string): CloudFormation.Template | undefined
-    addTemplateToTemplateData(templatePath: vscode.Uri): Promise<void>
-    removeTemplateFromRegistry(templatePath: vscode.Uri): void
-}
-
-export class DefaultCloudFormationTemplateRegistryListener implements FileWatcherListener {
-    public constructor(private readonly registry: CloudFormationTemplateRegistry) {}
-
-    /**
-     * Function used by a vscode.FileSystemWatcher to handle an onDidChange event.
-     * Adds template to data map. TODO: Emit event
-     * @param templateUri vscode.Uri for the changed template file
-     */
-    public async onListenedChange(templatePath: vscode.Uri): Promise<void> {
-        await this.registry.addTemplateToTemplateData(templatePath)
-        // TODO: fire event
-    }
-
-    /**
-     * Function used by a vscode.FileSystemWatcher to handle an onDidCreate event.
-     * Adds template to data map. TODO: Emit event
-     * @param templateUri vscode.Uri for the created template file
-     */
-    public async onListenedCreate(templatePath: vscode.Uri): Promise<void> {
-        await this.registry.addTemplateToTemplateData(templatePath)
-        // TODO: fire event
-    }
-
-    /**
-     * Function used by a vscode.FileSystemWatcher to handle an onDidDelete event.
-     * Removes template from data map. TODO: Emit event
-     * @param templateUri vscode.Uri for the deleted template file
-     */
-    public async onListenedDelete(templatePath: vscode.Uri): Promise<void> {
-        this.registry.removeTemplateFromRegistry(templatePath)
-        // TODO: fire event
-    }
-
-    /**
-     * Disposes the  vscode.EventEmitters tied to this class
-     * // TODO: Add event emitters to the class and dispose of them here
-     */
-    public dispose(): void {
-        // TODO: Add event emitters to be disposed
-    }
+    addTemplateToRegistry(templateUri: vscode.Uri): Promise<void>
+    addTemplatesToRegistry(templateUris: vscode.Uri[]): Promise<void>
+    removeTemplateFromRegistry(templateUri: vscode.Uri): void
+    reset(): void
 }
 
 export class DefaultCloudFormationTemplateRegistry implements CloudFormationTemplateRegistry {
```

Comment:
It should be possible to delete the interface `CloudFormationTemplateRegistry` and rename this class to `CloudFormationTemplateRegistry`. There won't be an alternate implementation, and this class contains simple functionality that can be tested in isolation.